### PR TITLE
Test the function of filter clean traffic gateway

### PR DESCRIPTION
**Pull request by yalzhang**
The nwfilter clean-traffic-gateway will block the vms connected to the
same bridge to access to each other.

Signed-off-by: yalzhang <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
depends on https://github.com/avocado-framework/avocado-vt/pull/2606

**Comment by yalzhang** [comment]
run locally pass
$ avocado run --vt-type libvirt nwfilter_vm_start.possitive_test.default_filter.clean_traffic_gateway
/root/code/avocado/avocado/plugins/run.py:294: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : b8ae597c21d6df4148d8c1d24596e7cb809b6ed1
JOB LOG    : /root/avocado/job-results/job-2020-08-06T00.01-b8ae597/job.log
 (1/1) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.clean_traffic_gateway: PASS (103.30 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 105.10 s

**Comment by yalzhang** [comment]
run all the cases locally and pass

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py [hunk 1]
```diff
@@ -98,6 +110,7 @@ def clean_up_dirty_nwfilter_binding():
         new_iface = interface.Interface('network')
         new_iface.xml = iface_xml.xml
         new_filterref = new_iface.new_filterref(**filterref_dict)
+        logging.debug("new_filterref.xml  is %s" % new_filterref.xml)
```
Comment (kylazhang) [inline libvirt/tests/src/nwfilter/nwfilter_vm_start.py:113]: Please check if it is needed, and if need, please clean the format and remove the redundant blank

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py [hunk 2]
```diff
@@ -157,6 +170,48 @@ def clean_up_dirty_nwfilter_binding():
                     test.fail("'%s' not found in output: %s"
                               % (expect_match, out))
 
+            if need_vm2:
+                # Clone more vm for testing
+                result = virsh.dom_list('--inactive').stdout_text
+                if "vm2" in result:
```
Comment (kylazhang) [inline libvirt/tests/src/nwfilter/nwfilter_vm_start.py:176]: I think the vm name should not be hard code in the scripts, or should get from the params? Also please deal with the following "vm2"s

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py [hunk 3]
```diff
@@ -39,6 +41,10 @@ def run(test, params, env):
     vm = env.get_vm(vm_name)
     username = params.get("username")
     password = params.get("password")
+    need_vm2 = "yes" == params.get("need_vm2", "no")
+    vms = [vm]
+    dst_outside = params.get("dst_outside", "www.google.com")
+    ping_timeout = float(params.get("ping_timeout", "10"))
```
Comment (chloerh) [inline libvirt/tests/src/nwfilter/nwfilter_vm_start.py:47]: Is 'int' better?

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/src/nwfilter/nwfilter_vm_start.py:176]: Agreed,  we should at least use a var to store 'vm2'

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py [hunk 4]
```diff
@@ -191,3 +246,12 @@ def clean_up_dirty_nwfilter_binding():
...
             process.run("ipset destroy blacklist", shell=True)
+        # Remove additional vms
+        if need_vm2 and vm2.exists():
+            virsh.remove_domain("vm2", "--remove-all-storage")
+
+        # Undefine newly renamed domain
+        if vm2.exists():
+            if vm2.is_alive():
+                vm2.destroy(gracefully=False)
+            vm2.undefine()
```
Comment (chloerh) [inline libvirt/tests/src/nwfilter/nwfilter_vm_start.py:257]: I'm confused, seems we are gonna remove 'vm2' anyway, but why do it twice and use different ways?

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py (hunk 1 above)
Comment (yalzhang) [reply libvirt/tests/src/nwfilter/nwfilter_vm_start.py]: deleted as it is the log I need during debug

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py (hunk 2 above)
Comment (yalzhang) [reply libvirt/tests/src/nwfilter/nwfilter_vm_start.py]: You are right, updated.

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py (hunk 3 above)
Comment (yalzhang) [reply libvirt/tests/src/nwfilter/nwfilter_vm_start.py]: Yes, updated

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py (hunk 4 above)
Comment (yalzhang) [reply libvirt/tests/src/nwfilter/nwfilter_vm_start.py]: oh, it is weird, I have updated and pushed again, Please review, Thank you!

**Comment by yalzhang** [comment]
Updated and pushed again, have run locally for all the cases for this file, no issues.

**Review by kylazhang — APPROVED** [review]

**Review by chloerh — APPROVED** [review]

**Comment by chloerh** [comment]
Dependency of vt has been merged.

**Pull request merged**